server/apolloServer.js

```diff
@@ -13,7 +14,13 @@ function newApolloServer (app, schema, httpServer, tracing, playgroundConfig, gr
     schema,
     validationRules: [
       queryDepthLimit(serverSecurity.queryDepthLimit),
-      createComplexityLimitRule(serverSecurity.complexityLimit)
+      createComplexityLimitRule(serverSecurity.complexityLimit, {
+        formatErrorMessage: (cost) => {
+          const errorMessage = `query with ${cost} exceeds complexity limit`
+          log.warn(errorMessage)
```

Comment:
I haven't tested this but I have a question. Can the admin see what query caused this log? Do we print the query in another log message before this one?

Reply:
@aliok not in its current form. I will take a look at the module and see if its possible to log the context or something similar.

Reply:
@aliok I forgot about this sorry. I just ran it with createError instead of formatErrorMessage which gives me access to the node. However I cannot customise it much beyond that. What that means is a huge log as every field gets a before and after cost printed to the console. Do you think a 400 error and the custom error back to the client is enough in this case?

Reply:
Ok, good point. We shouldn't be logging huge logs. That's a security risk itself vulnerable to flooding and DoS. Then it is fine as is.
Thanks @StephenCoady

Reply:
err, ok, one other option is to use "DEBUG" log level for logging that huge log. What do you think?

Reply:
actually that would work. I'll make those changes now.